packages/dm-core-plugins/src/header/HeaderPlugin.tsx

```diff
@@ -99,31 +102,42 @@ export default (props: IUIPlugin): JSX.Element => {
     return <Loading />
   }
 
+  const recipeNames =
+    config.uiRecipesList.length > 0
+      ? config.uiRecipesList
+      : uiRecipes.map((recipe: TUiRecipe) => recipe.name)
+
   return (
     <div>
-      <TopBar>
-        <TopBar.Header>
+      <TopBar
+        style={{
+          display: 'flex',
+          justifyItems: 'center',
+          justifyContent: 'space-between',
+        }}
+      >
+        <TopBar.Header style={{ position: 'relative' }}>
           <ClickableIcon
+            ref={setAnchorEl}
             onClick={() => {
               setAppSelectorOpen(!appSelectorOpen)
             }}
           >
-            <Icon data={grid_on} size={32} />
+            <Icon data={apps} size={32} />
           </ClickableIcon>
-          <h4 style={{ paddingTop: 9, paddingLeft: 10 }}>{entity.label}</h4>
-          {appSelectorOpen && (
-            <RecipeSelector
-              selectableUiRecipeNames={
-                config.uiRecipesList.length > 0
-                  ? config.uiRecipesList
-                  : uiRecipes.map((recipe: TUiRecipe) => recipe.name)
-              }
-              setSelectedUiRecipe={(uiRecipeName: string) => {
-                setAppSelectorOpen(false)
-                setSelectedRecipe(getRecipeConfigAndPlugin(uiRecipeName))
-              }}
-            />
-          )}
+          <Menu open={appSelectorOpen} anchorEl={anchorEl}>
+            {recipeNames.map((recipe, index) => (
+              <Menu.Item
+                key={index}
+                onClick={() =>
+                  setSelectedRecipe(getRecipeConfigAndPlugin(recipe))
```

Comment:
```suggestion
                  setSelectedRecipe(getRecipeConfigAndPlugin(recipe))
                  setAppSelectorOpen(false)
```
would be even better

Reply:
indeed!